### PR TITLE
A more POLA to_s on Collection

### PR DESCRIPTION
**Pull request by privat**
As announced in #1385, this PR change the default `to_s` on collections.

Before: `[1, 2, 3].to_s == "123"`
After: `[1, 2, 3].to_s == "[1,2,3]"`

The main issue is for code that used `to_s` on array to get a fast concatenation.
Those should now use `plain_to_s` (or `join("")`).

Close #off-line issue signaled by @ablondin

Note: this PR fails on bootstrap (`to_s` is used internally) thus requires that `c_src` is regenerated.

### CODE REVIEW

**Comment by jpages** [comment]
+1

File: lib/standard/string.nit [hunk 1]
```diff
@@ -2245,13 +2245,23 @@ redef class Char
 end
 
 redef class Collection[E]
-	# Concatenate elements.
+	# String representation of the content of the collection.
+	#
+	# The standard representation is the list of elements separated with comas.
```
Comment (Morriar) [inline lib/standard/string.nit]: commas*

File: lib/standard/string.nit [hunk 2]
```diff
@@ -2245,13 +2245,23 @@ redef class Char
...
+	# assert [1,2,3].to_s == "[1,2,3]"
+	# assert [1..3].to_s  == "[1,2,3]"
+	# assert (new Array[Int]).to_s == "[]" # empty collection
+	# ~~~
+	#
+	# Subclasses may return a more specific string representation.
 	redef fun to_s
 	do
-		return plain_to_s
+		return "[" + join(",") + "]"
```
Comment (Morriar) [inline lib/standard/string.nit]: No superstring?

**Comment by Morriar** [comment]
+1

**Comment by R4PaSs** [comment]
:shipit:

File: lib/standard/string.nit (hunk 2 above)
Comment (privat) [inline lib/standard/string.nit]: the interpreter caused some GC issues in some tests with superstring.
Once @R4PaSs job is completed, we will chose the most efficient

**Comment by privat** [comment]
reroll: six typo + add nitunits to plain_to_s

File: lib/standard/string.nit (hunk 2 above)
Comment (Morriar) [inline lib/standard/string.nit]: This is why we can't have nice things... :(

**Comment by privat** [comment]
rebase with new master to solve conflicts

**Pull request merged**